app/controllers/spam2_controller.rb

```diff
@@ -0,0 +1,132 @@
+class Spam2Controller < ApplicationController
+  before_action :require_user, only: %i(spam spam_revisions mark_comment_spam publish_comment spam_comments)
+
+  def _spam
+    if logged_in_as(%w(moderator admin))
+      @nodes = Node.paginate(page: params[:page])
+      @nodes = if params[:type] == 'wiki'
+                 @nodes.where(type: 'page', status: 1)
+               else
+                 @nodes.where(status: [0, 4])
+               end
+      @spam_count = @nodes.where(status: 0).length.to_s
+      @unmoderated_count = @nodes.where(status: 4).length.to_s
+    else
+      flash[:error] = 'Only moderators can moderate posts.'
+      redirect_to '/dashboard'
+    end
+  end
+
+  def _spam_revisions
+    if logged_in_as(%w(amdin moderator))
```

Comment:
please change `amdin` to `admin` here. Thanks :+1:

Reply:
Sorry for that, Changed it!